pkg/virt-launcher/virtwrap/device/hostdevice/sriov/hostdev.go

```diff
@@ -43,21 +43,22 @@ func CreateHostDevicesFromIfacesAndPool(ifaces []v1.Interface, pool hostdevice.A
 func createHostDevicesMetadata(ifaces []v1.Interface) []hostdevice.HostDeviceMetaData {
 	var hostDevicesMetaData []hostdevice.HostDeviceMetaData
 	for _, iface := range ifaces {
+		currentIface := iface
```

Comment:
If you leave the old variable, someone in the future may use it, causing this again.
I suggest you override it: `iface := iface`. This is a common pattern per what I saw.

Reply:
This will "shadow" it, and without an IDE you will not be able to tell the difference.
I suggest to add a comment why this is necessary instead, WDYT?

Reply:
It should shadow it, that is good.

The unit tests added cover the current fields, not future ones.

Reply:
I find the shadowing option confusing for a future reader.
I've added a comment which describes why this assignment is needed.

Reply:
I find this language gotcha confusing, and many more. But I am not going to teach readers this through such comments. Comments rot, I do not trust them.

Saying this on the commit message should be enough.

regarding the shadowing, I think in this case it is useful to lock missuse. We explicitly disable the optimization of reusing the same memory for the value.

Reply:
@ormergi I've went with your approach of adding a helper function.
Thanks for your suggestion.